Makefile

```diff
@@ -11,46 +11,60 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+include Makefile.common
+
 GO     ?= GO15VENDOREXPERIMENT=1 go
-GOPATH := $(firstword $(subst :, ,$(shell $(GO) env GOPATH)))
 GOARCH := $(shell $(GO) env GOARCH)
 GOHOSTARCH := $(shell $(GO) env GOHOSTARCH)
 
-PROMTOOL    ?= $(GOPATH)/bin/promtool
-ifeq ($(BUILD_PROMU),false)
-        PROMU       ?= promu
-else
-        PROMU       ?= $(GOPATH)/bin/promu
-endif
-STATICCHECK ?= $(GOPATH)/bin/staticcheck
-pkgs         = $(shell $(GO) list ./... | grep -v /vendor/)
+PROMTOOL_VERSION ?= 2.5.0
+PROMTOOL_URL     ?= https://github.com/prometheus/prometheus/releases/download/v$(PROMTOOL_VERSION)/prometheus-$(PROMTOOL_VERSION).$(GO_BUILD_PLATFORM).tar.gz
+PROMTOOL         ?= $(FIRST_GOPATH)/bin/promtool
 
-PREFIX                  ?= $(shell pwd)
-BIN_DIR                 ?= $(shell pwd)
 DOCKER_IMAGE_NAME       ?= node-exporter
-DOCKER_IMAGE_TAG        ?= $(subst /,-,$(shell git rev-parse --abbrev-ref HEAD))
 MACH                    ?= $(shell uname -m)
 DOCKERFILE              ?= Dockerfile
 
 STATICCHECK_IGNORE =
 
 ifeq ($(OS),Windows_NT)
-    OS_detected := Windows
+	OS_detected := Windows
 else
-    OS_detected := $(shell uname -s)
+	OS_detected := $(shell uname -s)
 endif
 
 ifeq ($(GOHOSTARCH),amd64)
 	ifeq ($(OS_detected),$(filter $(OS_detected),Linux FreeBSD Darwin Windows))
-                # Only supported on amd64
-                test-flags := -race
-        endif
+		# Only supported on amd64
+		test-flags := -race
+	endif
 endif
 
 ifeq ($(OS_detected), Linux)
-    test-e2e := test-e2e
+	test-e2e := test-e2e
+else
+	test-e2e := skip-test-e2e
+endif
+
+# Use CGO for non-Linux builds.
+ifeq ($(GOOS), linux)
+	PROMU_CONF ?= .promu.yml
+else
+	ifndef GOOS
+		ifeq ($(OS_detected), Linux)
+			PROMU_CONF ?= .promu.yml
+		else
+			PROMU_CONF ?= .promu-cgo.yml
+		endif
+	else
+		PROMU_CONF ?= .promu-cgo.yml
+	endif
+endif
+
+ifeq ($(GET_PROMU), false)
```

Comment:
Do we need this? AFAICT it isn't used anywhere, redundant with `BUILD_PROMU` and not present in upstream.

Reply:
Good catch, that should have been BUILD_PROMU, it's left over from my first attempt.  I guess it's not needed in this case because we don't care about the .promu-cgo.yml config.